hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -110,9 +111,11 @@ private[hudi] object HoodieSparkSqlWriter {
       if (!tableExists) {
         val archiveLogFolder = parameters.getOrElse(
           HoodieTableConfig.HOODIE_ARCHIVELOG_FOLDER_PROP_NAME, "archived")
+        val indexType = parameters.getOrElse(
```

Comment:
I think we have a place where all common properties are set.

    val parameters = HoodieWriterUtils.parametersWithWriteDefaults(optParams)
 
 can set the index type default there and simply use it here. i.e no need to do parameters.getOrElse, just parameters.get()